samtranslator/model/api/http_api_generator.py

```diff
@@ -164,6 +174,137 @@ def _add_cors(self):
         # Assign the OpenApi back to template
         self.definition_body = editor.openapi
 
+    def _construct_api_domain(self, http_api):
+        """
+        Constructs and returns the ApiGateway Domain and BasepathMapping
+        """
+        if self.domain is None:
+            return None, None, None
+
+        if self.domain.get("DomainName") is None or self.domain.get("CertificateArn") is None:
+            raise InvalidResourceException(
+                self.logical_id, "Custom Domains only works if both DomainName and CertificateArn" " are provided."
+            )
+
+        self.domain["ApiDomainName"] = "{}{}".format(
+            "ApiGatewayDomainNameV2", logical_id_generator.LogicalIdGenerator("", self.domain.get("DomainName")).gen()
+        )
+
+        domain = ApiGatewayV2DomainName(
+            self.domain.get("ApiDomainName"), attributes=self.passthrough_resource_attributes
+        )
```

Comment:
For the DomainName- since it can have tags, can we pass through the tags added at the API level to the domain?

Reply:
Not sure what you mean here.